cloud-hypervisor: add platformOEMStrings and `--platform` merging

Ramblurr opened the pull request:
The cloud-hypervisor command line interface unfortunately doesn't support
multiple instances of the same arg with a different value, so we have to resort
to these extra module options rather than using extraArgs.

To make matters even worse, the `--platform` argument (of which there can be
only one), is overloaded with different types of sub-args that also need to be
provided multiple times.

This commit allows the operator to add oem strings (for example to pass systemd
credentials), as well as raw platform options as needed.

Ramblurr commented on nixos-modules/microvm/options.nix:
```diff
@@ -483,6 +483,30 @@ in
       '';
     };
 
+    cloud-hypervisor.platformOEMStrings = mkOption {
```
That wouldn't quite parse it correctly because there is the square bracket list syntax too. And we have to combine it with the vsock OEM string. 

But we could parse them out and then rebuild the oem_strings arg, should I so that?

It's a shame that chv has such a clunky interface.

astro replied:
You're the one with the new use-case here :-)

If you agree that it is worthwhile working around chv's clunky interface, so please go ahead!

If you prefer separate options, I'm going to accept them.

Ramblurr replied:
IMO this is resolved.

Ramblurr commented:
Hi again @astro, I want to pick this little project back up.. so I'm going to be reviving these PRs today :) Please see my comments to your comments!.

I've just pushed new commits that should address your concerns.

SuperSandro2000 commented on lib/default.nix:
```diff
@@ -78,4 +78,59 @@ rec {
+      => { values = ["linux"]; args = ["-vnc" ":0" "-usb"]; }
+
+      # Extract multiple occurrences:
+      extractOptValues "-platform" ["-a" "a" "-b" "b" "-c" "c" "-b" "b2"]
```
```suggestion
      extractOptValues "-b" ["-a" "a" "-b" "b" "-c" "c" "-b" "b2"]
```

If I understand this correctly, the example should have ``-b`` instead.

SuperSandro2000 commented on lib/runners/cloud-hypervisor.nix:
```diff
@@ -5,8 +5,9 @@
 
 let
   inherit (pkgs) lib;
+  inherit (import ../. { inherit (pkgs) lib; }) extractOptValues;
```
I am not a fan of this, but this existed before in other files.

Ramblurr commented:
@SuperSandro2000 I've pushed a new commit that rebases onto `main` to pickup your latest PR, and also addresses your comments. Let me know if there is anything else.

Ramblurr commented:
fixed the merge conflict

SuperSandro2000 commented on lib/default.nix:
```diff
@@ -78,4 +78,59 @@ rec {
+
+      # Extract multiple occurrences:
+      extractOptValues "-b" ["-a" "a" "-b" "b" "-c" "c" "-b" "b2"]
+      => { values = ["b" "b2"]; args = ["-b" "b" "-c" "c"]; }
```
```suggestion
      => { values = ["b" "b2"]; args = ["-a" "a" "-c" "c"]; }
```

SuperSandro2000 commented:
Let's try this 🎉 thanks
